modules/porous_flow/src/materials/PorousFlowSingleComponentFluid.C

```diff
@@ -20,68 +24,100 @@ validParams<PorousFlowSingleComponentFluid>()
 
 PorousFlowSingleComponentFluid::PorousFlowSingleComponentFluid(const InputParameters & parameters)
   : PorousFlowFluidPropertiesBase(parameters),
-
-    _density(_nodal_material
-                 ? declareProperty<Real>("PorousFlow_fluid_phase_density_nodal" + _phase)
-                 : declareProperty<Real>("PorousFlow_fluid_phase_density_qp" + _phase)),
-    _ddensity_dp(_nodal_material
-                     ? declarePropertyDerivative<Real>(
-                           "PorousFlow_fluid_phase_density_nodal" + _phase, _pressure_variable_name)
-                     : declarePropertyDerivative<Real>("PorousFlow_fluid_phase_density_qp" + _phase,
-                                                       _pressure_variable_name)),
+    _compute_rho_nu(getParam<bool>("compute_density_and_viscosity")),
+    _compute_internal_energy(getParam<bool>("compute_internal_energy")),
+    _compute_enthalpy(getParam<bool>("compute_enthalpy")),
+    _density(_compute_rho_nu
+                 ? (_nodal_material
+                        ? &declareProperty<Real>("PorousFlow_fluid_phase_density_nodal" + _phase)
+                        : &declareProperty<Real>("PorousFlow_fluid_phase_density_qp" + _phase))
+                 : nullptr),
+    _ddensity_dp(
+        _compute_rho_nu
+            ? (_nodal_material
+                   ? &declarePropertyDerivative<Real>(
+                         "PorousFlow_fluid_phase_density_nodal" + _phase, _pressure_variable_name)
+                   : &declarePropertyDerivative<Real>("PorousFlow_fluid_phase_density_qp" + _phase,
+                                                      _pressure_variable_name))
```

Comment:
LOL - Oh clang format...  
BTW - If you find a particularly ugly piece of code like this you can put in a normal C++ comment to tell clang-format to leave it alone. That's up to you though.